typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -89,7 +99,7 @@ export abstract class HyperlaneDeployer<
   }
 
   cacheAddressesMap(addressesMap: HyperlaneAddressesMap<any>): void {
-    this.cachedAddresses = addressesMap;
+    objMerge(this.cachedAddresses, addressesMap);
```

Comment:
does this mutate?
I think you still need to assign to `this.cachedAddresses`

Reply:
fixed